### PR TITLE
CI:CTest results on cdash again

### PR DESCRIPTION
**Pull request by ViktorvdValk**

### CODE REVIEW

**Comment by mstaring** [comment]
lgtm

**Comment by N-Dekker** [comment]
I see 9 builds at the Experimental dashboard, last Friday: https://my.cdash.org/index.php?project=elastix&date=2021-02-20

They are from this pull request, right? Cool! However, it seems that the Build column is missing. Do you have a clue why?

**Comment by ViktorvdValk** [comment]
Yes, that was because I only included the Test results, I now included the build results as well. Is there more information we want to include?

See Cdash docs:

> CTest defines an ordered list of testing steps of which some or all may be run as a dashboard client:
> 
> Start
> Start a new dashboard submission to be composed of results recorded by the following steps. See the CTest Start Step section below.
> 
> Update
> Update the source tree from its version control repository. Record the old and new versions and the list of updated source files. See the CTest Update Step section below.
> 
> Configure
> Configure the software by running a command in the build tree. Record the configuration output log. See the CTest Configure Step section below.
> 
> Build
> Build the software by running a command in the build tree. Record the build output log and detect warnings and errors. See the CTest Build Step section below.
> 
> Test
> Test the software by loading a CTestTestfile.cmake from the build tree and executing the defined tests. Record the output and result of each test. See the CTest Test Step section below.
> 
> Coverage
> Compute coverage of the source code by running a coverage analysis tool and recording its output. See the CTest Coverage Step section below.
> 
> MemCheck
> Run the software test suite through a memory check tool. Record the test output, results, and issues reported by the tool. See the CTest MemCheck Step section below.
> 
> Submit
> Submit results recorded from other testing steps to the software quality dashboard server. See the CTest Submit Step section below.

**Comment by mstaring** [comment]
configure I guess also

**Comment by N-Dekker** [comment]
Ziet er al beter uit! https://my.cdash.org/index.php?project=elastix 

Wat ik nog mis is de naam van de branch of de PR die gebouwd is. Bij ITK staat die wel vermeld op hun dashboard: https://open.cdash.org/index.php?project=Insight

File: .github/workflows/ElastixGitHubActions.yml [hunk 1]
```diff
@@ -13,25 +13,26 @@ jobs:
...
       matrix:
-        os: [ubuntu-18.04, windows-2019, macos-10.15]
+#         os: [windows-2019]
+        os: [macos-10.15, ubuntu-18.04, windows-2019]
         include:
           - os: ubuntu-18.04
             c-compiler: "gcc"
             cxx-compiler: "g++"
-            itk-git-tag: "v5.1.2"
+            itk-git-tag: "0fce3ff86b9242a298a55a42337769eca8e8a4c8"
```
Comment (N-Dekker) [inline .github/workflows/ElastixGitHubActions.yml:22]: Maybe ITK v5.2rc03 ?

File: .github/workflows/ElastixGitHubActions.yml [hunk 2]
```diff
@@ -13,25 +13,26 @@ jobs:
     strategy:
       max-parallel: 3
       matrix:
-        os: [ubuntu-18.04, windows-2019, macos-10.15]
+#         os: [windows-2019]
```
Comment (N-Dekker) [inline .github/workflows/ElastixGitHubActions.yml:16]: I guess this line may be removed  😃

File: .github/workflows/ElastixGitHubActions.yml [hunk 3]
```diff
@@ -67,11 +68,15 @@ jobs:
...
+    - name: Get specific version of CMake, Ninja
+      uses: lukka/get-cmake@v3.18.3
+      
     - name: Download ITK
       if: steps.cache.outputs.cache-hit != 'true'
       run: |
-        git clone https://github.com/InsightSoftwareConsortium/ITK.git --branch ${{ matrix.itk-git-tag }} --depth 1 ITK-source
+        git clone https://github.com/InsightSoftwareConsortium/ITK.git ITK-source
+        cd ITK-source
+        git checkout ${{ matrix.itk-git-tag }}
```
Comment (N-Dekker) [inline .github/workflows/ElastixGitHubActions.yml:79]: Just wondering, did git argument  `--branch ${{ matrix.itk-git-tag }}` not work well? `--depth 1` might speed things up, but I'm not sure if it does, honestly.

File: .github/workflows/ElastixGitHubActions.yml (hunk 1 above)
Comment (ViktorvdValk) [reply .github/workflows/ElastixGitHubActions.yml:22]: That build still fails on the CI, Matt advised me to use a more recent one.

File: .github/workflows/ElastixGitHubActions.yml (hunk 3 above)
Comment (ViktorvdValk) [reply .github/workflows/ElastixGitHubActions.yml:79]: This doesn't work if you want to checkout a specific commit. (Both the --branch flag and the --depth flag)
Apparently git needs more depth to checkout commits. ;)

**Comment by ViktorvdValk** [comment]
Updated with #451

**Pull request closed without merging**